bep032tools/generator/tests/test_BEP032Generator.py

```diff
@@ -184,7 +184,56 @@ def test_generate_example_structure(self):
     def doCleanups(self):
         initialize_test_directory(clean=True)
 
+class Test_FolderGeneration(unittest.TestCase):
+
+    def setUp(self):
+        self.test_dir = Path(initialize_test_directory(clean=True))
+        os.mkdir(self.test_dir / 'generateTest')
+        csv_filename = generate_simple_csv_file()
+        self.csv_file = csv_filename
+
+    def test_generate_folder(self):
+        """
+        When the file did not exist, the generation did not work properly.
+        Now if the folder does not exist, the function creates the folder before the generation function is launched
+        Parameters
+        ----------
+        Returns
+        ----------
+        bool
+            True if the generation is well done. False if it failed.
+        """
+        self.test_dir = self.test_dir / 'test'
+        test_generate = False
+        if os.path.isdir(self.test_dir):
+            BEP032Data.generate_struct(self.csv_file, self.test_dir)
+            test_generate = True
+        else:
+            os.mkdir(self.test_dir)
+            BEP032Data.generate_struct(self.csv_file, self.test_dir)
+            test_generate = True
+        self.assertTrue(test_generate)
+
+    def test_no_duplicate_folder_generated(self):
+        """
+        Checks that there are no duplicates when generating folders.
+        If a duplicate exists and has the same name as the parent folder, the function returns an error
+        Parameters
+        ----------
+        Returns
+        ----------
+        bool
+            True if there are no duplicates. False if a duplicate exists
+        """
+        generation = False
+        if os.path.isdir(self.test_dir) and os.path.dirname(self.test_dir) not in os.listdir(self.test_dir):
+            generation = True
+        self.assertTrue(generation)
```

Comment:
Yes, this is getting closer to what the test should look like. You have code to check for the existence of folders, the only thing that is missing is that you actually run the method that you want to test in your test. You need to call `BEP032Data.generate_struct(self.csv_file, self.test_dir)` somewhere before checking that no duplicate subfolders are generated.